src/vs/platform/actions/common/actions.ts

```diff
@@ -238,7 +238,7 @@ export interface IMenuService {
 	 * submenu entries. That is more expensive and must be explicitly enabled with the
 	 * `emitEventsForSubmenuChanges` flag.
 	 */
-	createMenu(id: MenuId, contextKeyService: IContextKeyService, options?: IMenuCreateOptions): IMenu;
+	createMenu(id: MenuId, contextKeyService: IContextKeyService | IOverlayContextKeyService, options?: IMenuCreateOptions): IMenu;
```

Comment:
This union type is a bit ugly but it seemed more conservative to update the few places where it's beneficial to be able to drop in an overlay context key service than to update all places that reference `IContextKeyService`

Reply:
Maybe `IContextKeyService` needn't have a `dispose()` method at all... but simply its implementations should. 🤔

Reply:
If I'm understanding what you're saying, would this lead to context key service instances not being disposed because the `dispose` method isn't advertised on its public interface?

Reply:
Not really... `ContextKeyService` instances would still have `dispose()`, so whoever instantiates them is responsible for disposing them. It's rather that the `IContextKeyService` interface needn't be disposable. In other words, one could conceptually create an implementation of `IContextKeyService` which has no need for disposable... and in fact that's exactly what the overlay context key service appears to be.

Reply:
Yea that makes sense. The scoped context key service does need to be disposed though so I introduced a new interface for that, LMK what you think of the current approach.